docker/entrypoint.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+# Correcting the URL format
+URL=$(echo "$URL" | sed 's|/$||')  # Remove trailing slash if exists
+if ! echo "$URL" | grep -q "^https://"; then
+  URL="https://$URL"  # Add https:// if not present
+fi
+
+# Exporting the corrected URL
+export URL
+
+# Run pikaraoke with environment variables
+if [ -z "$PASSWORD" ]; then
+  figlet PiKaraoke
+  /pikaraoke/pikaraoke.sh -d /pikaraoke-songs/ --headless -u $URL
```

Comment:
Is it possible for $URL to be optional? I was trying this out and having issues trying to get a basic localhost instance running. It wants a URL, but inputting http://localhost:5555 seems redundant since the omission of -u would do that

Reply:
It can be, but with the way docker environments work, it will make the QR code function break. Docker environments network wise are like little subnetworks that sit beneath the network the docker host is sitting on. The spun up containers are almost like VM's because they all have their own IP within the docker subnet, and each container can be exposing their services on the same port without conflicting with one another because they are on their own individual ip.

By default, most containers will assign a randomized string of letters or numbers to their hostname, so without the URL flag, the QR code will read out something like ab362c89:5555 which will be completely unreachable to the end users. 

If you wanted to have the end users reach the docker host at 5555 on a local network, that would work, but I feel like most people who will be spinning up a containerized version of this project like me know that they will be using a reverse proxy to route an internet domain, or localized fqdn to the containers services.

Reply:
Super helpful info. I noticed it was putting in a blank or otherwise invalid hostname. Yes, I think if you're running locally, not using Docker would be simpler overall.

Reply:
What I might do is edit the coding so URL can be left blank, and if it is, the bash scripting will determine the docker hosts hostname and input this data for the user... so local hosting can still occur.

The resulting QR code would then read dockerhostname:5555 or dockerhostname.domain.local:5555